### PR TITLE
ref: type src/sentry/auth/system.py

### PR DESCRIPTION
**Pull request by asottile-sentry**

### CODE REVIEW

File: src/sentry/auth/access.py [hunk 1]
```diff
@@ -993,6 +993,7 @@ def from_api_member(
 def from_auth(auth: ApiKey | SystemToken, organization: Organization) -> Access:
     if is_system_auth(auth):
         return SystemAccess()
+    assert not isinstance(auth, SystemToken)
```
Comment (evanh) [inline src/sentry/auth/access.py:996]: Are you sure this assert statement will never fail? Or is the AssertionError handled somewhere else?

File: src/sentry/auth/system.py [hunk 2]
```diff
@@ -14,17 +18,17 @@
...
 
 
-def get_system_token():
+def get_system_token() -> str:
     token = options.get("sentry:system-token")
     if not token:
         token = uuid4().hex
         options.set("sentry:system-token", token)
-    return token
+    return token  # type: ignore[no-any-return]
```
Comment (evanh) [inline src/sentry/auth/system.py:31]: Why does this need a `# type: ignore`?

File: src/sentry/auth/access.py (hunk 1 above)
Comment (asottile-sentry) [reply src/sentry/auth/access.py:996]: `is_system_auth` is implemented as `isinstance(auth, SystemToken)` -- the type checker would understand this narrowing except `ApiKey` is currently `Any`

File: src/sentry/auth/system.py (hunk 2 above)
Comment (asottile-sentry) [reply src/sentry/auth/system.py:31]: the type checker does not understand django options so (as the type ignore comment says) it does not allow `Any` to be returned when a type is specified

**Review by evanh — APPROVED** [review]

**Pull request merged**